opendevin/sandbox/ssh_box.py

```diff
@@ -206,6 +211,28 @@ def execute(self, cmd: str) -> Tuple[int, str]:
         exit_code = int(exit_code.lstrip('echo $?').strip())
         return exit_code, command_output
 
+    def setup_jupyter(self):
```

Comment:
Instead of putting this into SSHBox, could we have a `JupyterBox` that extends SSHBox?

Reply:
I am also debating about this: Ideally, the Python execution capability should be *parallel* to bash execution so that every agent can resort to one of them when needed. In that case, I think it probably makes sense for us to extend the existing `SSHBox` to have a separate Python execution capability via `execute_python` as it won't interfere with the existing actions going through the `SSHBox` and other existing agents can also have `python` execution natively supported easily without specifying a different `SSHJupyterBox`?

Or we can just do the sub-class and set the `SSHJupyterBox` to the default just to make the code look better :) - anyway, I can create a sub-class first, and then we can decide whether we want this Python execution capability to be a default behavior.